Source/Diagnostics/ComputeDiagFunctors/TemperatureFunctor.cpp

```diff
@@ -53,22 +54,10 @@ TemperatureFunctor::operator() (amrex::MultiFab& mf_dst, const int dcomp, const
                 amrex::GpuArray<amrex::Real,AMREX_SPACEDIM> const& dxi)
             {
                 // Get position in AMReX convention to calculate corresponding index.
-                // Ideally this will be replaced with the AMReX NGP interpolator
-                // Always do x direction.
-                int ii = 0, jj = 0, kk = 0;
-                const amrex::ParticleReal x = p.pos(0);
-                const amrex::Real lx = (x - plo[0]) * dxi[0];
-                ii = static_cast<int>(amrex::Math::floor(lx));
-#if defined(WARPX_DIM_XZ) || defined(WARPX_DIM_3D) || defined(WARPX_DIM_RZ)
-                const amrex::ParticleReal y = p.pos(1);
-                const amrex::Real ly = (y - plo[1]) * dxi[1];
-                jj = static_cast<int>(amrex::Math::floor(ly));
-#endif
-#if defined(WARPX_DIM_3D)
-                const amrex::ParticleReal z = p.pos(2);
-                const amrex::Real lz = (z - plo[2]) * dxi[2];
-                kk = static_cast<int>(amrex::Math::floor(lz));
-#endif
+                const auto cellIndex = ParticleUtils::getParticleCellIndex(p.pos(0), p.pos(1), p.pos(2), plo, dxi, amrex::Dim3{0, 0, 0}).dim3();
```

Comment:
Here you could simply do the following, right?
```suggestion
                const auto cellIndex = ParticleUtils::getParticleCellIndex(p, plo, dxi, amrex::Dim3{0, 0, 0}).dim3();
```
Which is a bit cleaner.

Reply:
Here, there was an issue in that in `TemperatureFunctor`, `p` is a `WarpXParticleContainer::SuperParticleType` but the function expects `ParticleType`.  The opposite is true in `ParticleUtils.cpp`, where a `ParticleType` is used. I couldn't find definitions for these types, but clang warned me there was no way to cast between them.

Reply:
Oh, I see. Thanks for clarifying!

Reply:
I could either 1) remove the overload for `ParticleType` so we just pass in the coords in both locations or 2) add an overload for `SuperParticleType` if you think that would be valuable

Reply:
Adding another overload for `SuperParticleType` can't hurt, so I vote for that :)
Thanks!